### PR TITLE
chore: separate fedimint config info from gateway info endpoint

### LINKED ISSUE #3815: `gateway-cli info` displays far too much information
It includes the entire client config. This is a problem because when you run `gateway-cli info` you can't see at a glance how many federations you're in.

### PR DESCRIPTION
**Pull request by mayrf**
Fixes #3815. I am not sure it's completely done and would appreciate feedback.

### CODE REVIEW

File: gateway/cli/src/main.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ pub enum Commands {
     VersionHash,
     /// Display high-level information about the Gateway
     Info,
+    /// Display config information about the Gateways federation
+    Config,
```
Comment (elsirion) [inline gateway/cli/src/main.rs:34]: Imo this should be a per-federation command, i.e. takes the federation id as argument.

**Review by elsirion — COMMENTED** [review]
Looks good, but I think we can simplify it a bit more.

File: gateway/ln-gateway/src/lib.rs [hunk 2]
```diff
@@ -1217,24 +1233,37 @@ impl Gateway {
...
+        federation_id: FederationId,
+    ) -> FederationConfigInfo {
+        let balance_msat = client.get_balance().await;
+        let config = client.get_config().clone();
+
+        FederationConfigInfo {
+            federation_id,
+            balance_msat,
+            config,
+        }
```
[the remark this answers is not in the document]
Comment (dpc) [reply gateway/ln-gateway/src/lib.rs:1248]: If we remove "balance", ... should it be just a map? id -> config ?

File: gateway/cli/src/main.rs (hunk 1 above)
Comment (dpc) [reply gateway/cli/src/main.rs:34]: I wonder which is better. This or `--federation-id` acting as a filter, and lack of it meaning "all".

Comment (elsirion) [reply gateway/cli/src/main.rs:34]: Why do we need the "all" version of the command?

Comment (dpc) [reply gateway/cli/src/main.rs:34]: I don't know. If you're in one federation only you can get the config without specifying the id. Or if you're writing a script you could loop over all in one go (one `| jq ...`). Not very strong reasons, but also doesn't change anything for someone that just wants one config.

File: gateway/ln-gateway/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dpc) [reply gateway/ln-gateway/src/lib.rs:1248]: I guess it's a follow up to do we allow listing all or not. If we do , will be easier to parse/filter, if not it's kind of needless.

**Comment by mayrf** [comment]
Sorry for the delay. I went for dpcs suggestion and implemented an --federation-id option, returning either the config of the federation specified or those of all federations, in case no federation-id is specified.

File: gateway/ln-gateway/src/rpc/rpc_client.rs [hunk 3]
```diff
@@ -94,7 +99,7 @@ impl GatewayRpcClient {
             .base_url
             .join("/set_configuration")
             .expect("invalid base url");
-        self.call(Method::POST, url, Some(payload)).await
+        self.call_post(url, payload).await
```
Comment (elsirion) [inline gateway/ln-gateway/src/rpc/rpc_client.rs:102]: Do we want to have composite calls in fedimint-cli? Can be fixed in another PR.

**Review by elsirion — APPROVED** [review]

File: gateway/ln-gateway/src/rpc/mod.rs [hunk 4]
```diff
@@ -82,6 +92,12 @@ pub struct GatewayInfo {
     pub network: Option<Network>,
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct GatewayFedConfig {
+    pub federations: BTreeMap<FederationId, JsonClientConfig>,
+    // pub federations: BTreeMap<FederationId, ClientConfig>,
```
Comment (okjodom) [inline gateway/ln-gateway/src/rpc/mod.rs:98]: nit: let's clean up commented code

**Review by okjodom — APPROVED** [review]
left nit comments, yet this looks and works great!

tACK https://github.com/fedimint/fedimint/pull/3880/commits/1bc5bfd33cd65d9975775d11e56623ee93193b25

**Review by okjodom — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

**Pull request merged**